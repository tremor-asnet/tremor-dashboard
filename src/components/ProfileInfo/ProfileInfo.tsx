"use client";

// Libs
import { usePathname } from "next/navigation";

// Components
import Avatar from "../../components/Avatar/Avatar";
import { Title, Text } from "@tremor/react";

// Constants
import { ROUTES } from "../../constants";

interface ProfileInfoProps {
  isOnHeader: boolean;
  name: string;
  role: string;
  src: string;
}

const ProfileInfo = ({
  isOnHeader = true,
  name,
  role,
  src = "/images/avatar/avatar-lg.webp",
}: ProfileInfoProps) => {
  const pathname = usePathname();
  const isProjectPage = pathname === ROUTES.PROJECTS;

  return (
    <div className="flex flex-col lg:flex-row lg:items-center justify-between px-4 py-1">
      <div
        className={`flex flex-col xs:flex-row xs:items-center profile-info ${
          isProjectPage ? "mb-0" : "mb-6"
        }`}>
        <div className="flex items-center">
          {isOnHeader ? (
            <Avatar
              alt="Avatar medium"
              className="shadow-lg"
              height={74}
              priority
              src={src}
              width={74}
            />
          ) : (
            <Avatar
              alt="Avatar medium"
              className="shadow-md"
              height={48}
              priority
              src={src}
              width={48}
            />
          )}
        </div>
        {isOnHeader ? (
          <div className="xs:ml-6" data-testid="profile-info-lg">
            <Title className="font-semibold text-xl">{name}</Title>
            <Text>{role}</Text>
          </div>
        ) : (
          <div className="ml-4" data-testid="profile-info-md">
            <Title className="font-semibold">{name}</Title>
            <Text className="text-xs font-light truncate max-w-[165px]">
              {role}
            </Text>
          </div>
        )}
      </div>
    </div>
  );
};

export default ProfileInfo;
