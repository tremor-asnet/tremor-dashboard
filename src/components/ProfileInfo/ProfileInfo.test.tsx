import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import ProfileInfo from "./ProfileInfo";

describe("Testing profile info component", () => {
  const propsDefault = {
    isOnHeader: true,
    name: "Richard Davis",
    info: "CEO / Co-Founder",
    src: "/images/avatar/avatar-lg.webp",
  };

  it("Should render correctly name when isOnHeader is true", () => {
    const { container } = render(<ProfileInfo {...propsDefault} />);
    const profileInfoLg = container.getElementsByClassName("ml-6")[0];

    expect(profileInfoLg.firstChild?.textContent).toEqual("Richard Davis");
  });

  it("Should render correctly name when isOnHeader is false", () => {
    const { container } = render(
      <ProfileInfo {...propsDefault} isOnHeader={false} />,
    );
    const profileInfoMd = screen.getByTestId("profile-info-md");

    expect(profileInfoMd.firstChild).toHaveClass("font-semibold");
  });

  it("Should match snapshot", () => {
    const component = render(<ProfileInfo {...propsDefault} />);
    expect(component).toMatchSnapshot();
  });
});
