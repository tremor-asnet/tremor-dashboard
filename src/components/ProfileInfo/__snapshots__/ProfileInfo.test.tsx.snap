// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing profile info component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col lg:flex-row lg:items-center justify-between px-4 py-1"
      >
        <div
          class="flex flex-col md:flex-row md:items-center profile-info mb-6"
        >
          <div
            class="flex items-center"
          >
            <div
              class="inline-flex rounded-full overflow-hidden shadow-lg"
            >
              <img
                alt="Avatar medium"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="74"
                src="/_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=96&q=75 1x, /_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=256&q=75 2x"
                style="color: transparent;"
                width="74"
              />
            </div>
          </div>
          <div
            class="md:ml-6"
            data-testid="profile-info-lg"
          >
            <p
              class="text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis font-semibold text-xl"
            >
              Richard Davis
            </p>
            <p
              class="text-tremor-default text-tremor-content dark:text-dark-tremor-content"
            >
              CEO / Co-Founder
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col lg:flex-row lg:items-center justify-between px-4 py-1"
    >
      <div
        class="flex flex-col md:flex-row md:items-center profile-info mb-6"
      >
        <div
          class="flex items-center"
        >
          <div
            class="inline-flex rounded-full overflow-hidden shadow-lg"
          >
            <img
              alt="Avatar medium"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="74"
              src="/_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=256&q=75"
              srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=96&q=75 1x, /_next/image?url=%2Fimages%2Favatar%2Favatar-lg.webp&w=256&q=75 2x"
              style="color: transparent;"
              width="74"
            />
          </div>
        </div>
        <div
          class="md:ml-6"
          data-testid="profile-info-lg"
        >
          <p
            class="text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis font-semibold text-xl"
          >
            Richard Davis
          </p>
          <p
            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content"
          >
            CEO / Co-Founder
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
